### ISSUE TITLE
TypeError when same ref is used twice

### ISSUE THREAD

**Issue by anantoghosh** [cellular/oazapfts#31]
Using oazapfts@3.1.0
node: 12 lts
windows

api.yaml - here the two schemas reference the same `type.yaml` file twice
```yaml
openapi: 3.0.3
info:
  title: API
  version: 0.1.0
paths:
  "/url":
    get:
      parameters:
        - name: channel
          schema:
            $ref: "./type.yaml"
          in: query
        - name: number
          schema:
            $ref: "./type.yaml"
          in: query
      responses:
        "200":
          description: ok
```
type.yaml
```yaml
type: number
```

Then `oazapfts` crashes with
```bash
(node:29664) UnhandledPromiseRejectionWarning: TypeError: Cannot read property 'title' of undefined
    at getRefAlias (C:\Work\ip-ocd-ui\node_modules\.pnpm\oazapfts@3.1.0\node_modules\oazapfts\lib\codegen\generate.js:197:33)
    at getBaseTypeFromSchema (C:\Work\ip-ocd-ui\node_modules\.pnpm\oazapfts@3.1.0\node_modules\oazapfts\lib\codegen\generate.js:227:20)
    at getTypeFromSchema (C:\Work\ip-ocd-ui\node_modules\.pnpm\oazapfts@3.1.0\node_modules\oazapfts\lib\codegen\generate.js:214:22)
    at C:\Work\ip-ocd-ui\node_modules\.pnpm\oazapfts@3.1.0\node_modules\oazapfts\lib\codegen\generate.js:278:23
    at Array.map (<anonymous>)
    at getTypeFromProperties (C:\Work\ip-ocd-ui\node_modules\.pnpm\oazapfts@3.1.0\node_modules\oazapfts\lib\codegen\generate.js:272:44)
    at getBaseTypeFromSchema (C:\Work\ip-ocd-ui\node_modules\.pnpm\oazapfts@3.1.0\node_modules\oazapfts\lib\codegen\generate.js:247:20)
    at getTypeFromSchema (C:\Work\ip-ocd-ui\node_modules\.pnpm\oazapfts@3.1.0\node_modules\oazapfts\lib\codegen\generate.js:214:22)
    at C:\Work\ip-ocd-ui\node_modules\.pnpm\oazapfts@3.1.0\node_modules\oazapfts\lib\codegen\generate.js:278:23
    at Array.map (<anonymous>)
(node:29664) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 1)
(node:29664) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
```

I noticed that when generating the bundled spec file via `swagger-cli` the second `$ref` references the first type
```yaml
openapi: 3.0.3
info:
  title: API
  version: 0.1.0
paths:
  /url:
    get:
      parameters:
        - name: channel
          schema:
            type: number
          in: query
        - name: number
          schema:
            $ref: '#/paths/~1url/get/parameters/0/schema'
          in: query
      responses:
        '200':
          description: ok

```

It works fine when refs are different i.e. bundled spec file does not use this format of reference.

**Comment by anantoghosh** [comment]
As a workaround, we are now using ` -r, --dereference Fully dereference all $ref pointers` option of the swagger cli
related https://apitools.dev/swagger-parser/docs/swagger-parser.html#dereferenceapi-options-callback
to first create a bundled spec then running oazapfts on it.

**Comment by fgnass** [comment]
Fixed in oazapfts@3.1.1

**Comment by anantoghosh** [comment]
Hi @fgnass thank you for fixing this issue the minimal case is working now.

But when a path parameter is introduced, it starts crashing again
(same as previous example with {channel} added in the path)
```yaml
openapi: 3.0.3
info:
  title: API
  version: 0.1.0
paths:
  "/url/{channel}":
    get:
      parameters:
        - name: channel
          schema:
            $ref: "./type.yaml"
          in: path
          required: true
        - name: number
          schema:
            $ref: "./type.yaml"
          in: query
      responses:
        "200":
          description: ok
```

here the second ref was generated as
`"$ref": "#/paths/~1url~1%7Bchannel%7D/get/parameters/0/schema"`

**Comment by fgnass** [comment]
Good catch, fixed in oazapfts@3.1.2

**Comment by anantoghosh** [comment]
Thank you @fgnass  👑 !
